### PR TITLE
Merge 15919, 15897, 15763, 16912, 17001, 16656, 16804

### PR DESCRIPTION
**Pull request by vijaydasmp**

### CODE REVIEW

**Comment by vijaydasmp** [comment]
bitcoin#14987 This is Vlad_staged just picked up the changes required to merge Bitcoin#15620

**Comment by dzutte-cpp** [comment]
@vijaydasmp bitcoin#14987 is backported here: https://github.com/dashpay/dash/pull/4567
Feel free to rebase your PR once the backport is in.

**Comment by dzutte-cpp** [comment]
FYI - https://github.com/dashpay/dash/pull/4567 is checked in.

**Comment by vijaydasmp** [comment]
> FYI - #4567 is checked in.

Thanks

**Comment by PastaPastaPasta** [comment]
Please change `(Partial) Merge` -> `partial Merge`

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -432,7 +432,9 @@ static UniValue getdifficulty(const JSONRPCRequest& request)
 
 static std::string EntryDescriptionString()
 {
-    return "    \"size\" : n,                 (numeric) transaction size in bytes\n"
+    return "    \"vsize\" : n,                (numeric) transaction size in bytes\n"
```
Comment (PastaPastaPasta) [inline src/rpc/blockchain.cpp:435]: I feel like we should just drop 15637. We don't really have vsize, and not sure it makes sense to add it

File: src/test/crypto_tests.cpp [hunk 2]
```diff
@@ -21,7 +21,6 @@
 #include <vector>
 
 #include <boost/test/unit_test.hpp>
-#include <openssl/aes.h>
 #include <openssl/evp.h>
```
Comment (PastaPastaPasta) [inline src/test/crypto_tests.cpp:24]: upstream dropped evp.h should we not? (I presume it may be used, but want some clarity)

File: doc/bips.md [hunk 3]
```diff
@@ -15,11 +15,11 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.13.0**):
...
 * [`BIP 68`](https://github.com/bitcoin/bips/blob/master/bip-0068.mediawiki): Sequence locks have been implemented as of **v0.12.1**  ([PR #7184](https://github.com/bitcoin/bitcoin/pull/7184)), and have been activated since *block 419328*.
-* [`BIP 70`](https://github.com/bitcoin/bips/blob/master/bip-0070.mediawiki) [`71`](https://github.com/bitcoin/bips/blob/master/bip-0071.mediawiki) [`72`](https://github.com/bitcoin/bips/blob/master/bip-0072.mediawiki): Payment Protocol support has been available in Bitcoin Core GUI since **v0.9.0** ([PR #5216](https://github.com/bitcoin/bitcoin/pull/5216)).
+* [`BIP 70`](https://github.com/bitcoin/bips/blob/master/bip-0070.mediawiki) [`71`](https://github.com/bitcoin/bips/blob/master/bip-0071.mediawiki) [`72`](https://github.com/bitcoin/bips/blob/master/bip-0072.mediawiki): Payment Protocol support has been available in Bitcoin Core GUI since **v0.9.0** ([PR #5216](https://github.com/bitcoin/bitcoin/pull/5216)). Support can be optionally disabled at build time since **v0.18.0** ([PR 14451](https://github.com/bitcoin/bitcoin/pull/14451)).
```
Comment (PastaPastaPasta) [inline doc/bips.md:22]: We should probably link our PR here?

File: doc/bips.md [hunk 4]
```diff
@@ -15,11 +15,11 @@ BIPs that are implemented by Bitcoin Core (up-to-date up to **v0.13.0**):
...
 * [`BIP 42`](https://github.com/bitcoin/bips/blob/master/bip-0042.mediawiki): The bug that would have caused the subsidy schedule to resume after block 13440000 was fixed in **v0.9.2** ([PR #3842](https://github.com/bitcoin/bitcoin/pull/3842)).
-* [`BIP 61`](https://github.com/bitcoin/bips/blob/master/bip-0061.mediawiki): The 'reject' protocol message (and the protocol version bump to 70002) was added in **v0.9.0** ([PR #3185](https://github.com/bitcoin/bitcoin/pull/3185)). Starting *v0.16.0*, whether to send reject messages can be configured with the `-enablebip61` option.
+* [`BIP 61`](https://github.com/bitcoin/bips/blob/master/bip-0061.mediawiki): The 'reject' protocol message (and the protocol version bump to 70002) was added in **v0.9.0** ([PR #3185](https://github.com/bitcoin/bitcoin/pull/3185)). Starting **v0.17.0**, whether to send reject messages can be configured with the `-enablebip61` option, and support is deprecated as of **v0.18.0**.
```
Comment (PastaPastaPasta) [inline doc/bips.md:18]: support has not been deprecated

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/rpc/blockchain.cpp (hunk 1 above)
Comment (UdjinM6) [reply src/rpc/blockchain.cpp:435]: Note: We kind of have vsize internally now after #4562 but we don't have it in any rpcs, so 15637 is still a DNM probably

Comment (vijaydasmp) [reply src/rpc/blockchain.cpp:435]: Dropped 15637

File: src/test/crypto_tests.cpp (hunk 2 above)
Comment (vijaydasmp) [reply src/test/crypto_tests.cpp:24]: In our src/test/crypto_tests.cpp at BOOST_AUTO_TEST_CASE(pbkdf2_hmac_sha512_test) we have EVP_sha512  requiring evp.h. upstream doesn't have this testcase

**Comment by vijaydasmp** [comment]
Hello @PastaPastaPasta  @UdjinM6 , I have updated the PR , Please have a look

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: doc/README_doxygen.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
+\mainpage notitle
+
+\section intro_sec Introduction
+
+This is the developer documentation of the reference client for an experimental new digital currency called Bitcoin,
```
Comment (PastaPastaPasta) [inline doc/README_doxygen.md:5]: need to dashify this

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: doc/Doxyfile.in [hunk 6]
```diff
@@ -989,7 +989,7 @@ FILTER_SOURCE_PATTERNS =
 # (index.html). This can be useful if you have a project on for instance GitHub
 # and want to reuse the introduction page also for the doxygen output.
 
-USE_MDFILE_AS_MAINPAGE =
+USE_MDFILE_AS_MAINPAGE = doc/README_doxygen.md
```
[the remark this answers is not in the document]
Comment (vijaydasmp) [reply doc/Doxyfile.in:992]: implemented the Review comment

**Review by UdjinM6 — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Pull request merged**